crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1133,6 +1162,92 @@ impl PgManager {
             protocol_version: cfg.protocol_version.as_u64(),
         })
     }
+
+    pub(crate) async fn fetch_events(
+        &self,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+        filter: EventFilter,
+    ) -> Result<Option<Connection<String, Event>>, Error> {
+        let event_filter: Result<RpcEventFilter, Error> = if let Some(sender) = filter.sender {
```

Comment:
this is pretty complex, we'd probably want to back it up with some proper tests later.